x-pack/docs/en/security/authentication/security-domain.asciidoc

```diff
@@ -0,0 +1,98 @@
+[role="xpack"]
+[[security-domain]]
+=== Security domains
+
+beta::[]
+
+Security domains are a method of grouping different <<realms,realms>>
+so that the {stack} can recognize when a single user authenticates with
+these realms.
+
+==== How security domains work
+
+Some types of resources in {es} are owned by a single user.
+For example <<async-search,async search contexts>>,
+<<security-api-create-api-key,API keys>>,
+and <<user-profile,user profiles>> each have an owning user.
+
+When a user creates a resource, {es} captures the user's `username` and realm
+information as part of the resource's metadata.
+When the user attempts to access the resource at a later time, {es} compares
+the captured `username` and realm information against those from the accessing
+user. The access is not allowed unless both the realm and `username` match.
+That is, if {es} detects that a `username` from two different realms is
+attempting to access a resource, {es} assumes that these users are distinct and
+doesn't allow resources to be shared between them.
+
+However, there are cases where the same user can authenticate with
+different realms and need to share the same set of resources across the realms.
+For example, an <<ldap-realm,LDAP realm>> and a <<saml-realm,SAML realm>> can
+be backed by the same directory service. Additionally,
+<<configuring-authorization-delegation,authorization delegation>> allows one
+realm to delegate authorization to another realm. If both realms authenticate
+users with the same `username`, it is reasonable to treat these users to
+be the same user.
+
+With security domains, you can configure multiple realms to be under the same
+domain, users can authenticate with any of the realms and have access to the
+same set of resources regardless of which realm authenticating from,
+(provided those realms are under the same domain).
+For example, a single <<user-profile,user profile>> is
+associated to the user, enabling user preferences, notifications, and other
+user data to be shared across domains. The same user can view results from an
+asynchronous search request or a scrolling search. If the user has the
+necessary privileges, they can also view and manage API keys across realms.
+
+==== Configure a security domain
+
+You specify the realms for a domain in `elasticsearch.yml`. The realms must
+be defined in `elasticsearch.yml` (but do not need to be enabled).
+Specifying either an undefined realm, synthetic realm, or the reserved realm
+under a domain will prevent the node from starting.
+A realm doesn't need to belong to a domain. If it does, the realm can only
+belong to a single domain. Specifying a realm in multiple domains generates an
+error when {es} starts.
```

Comment:
```suggestion
Realms can be associated to at most one domain. It is an error ({es} fails to start) to assign an undefined realm to a domain or to assign a single realm to multiple domains.
```

Reply:
I reordered this section and moved it under the configuration example which has a second step says "Restart ES". It feels more naturally to talk about what could cause failure to start right after the instruction of restart. It also reads better to give example first before talking about error conditions.